src/nunit.analyzers/TestCaseSourceUsage/TestCaseSourceUsesStringAnalyzer.cs

```diff
@@ -43,22 +43,41 @@ private void AnalyzeAttribute(SyntaxNodeAnalysisContext context)
             {
                 context.CancellationToken.ThrowIfCancellationRequested();
 
-                var arguments = attributeNode.ArgumentList.Arguments;
+                if (attributeNode.ArgumentList is AttributeArgumentListSyntax argumentList &&

```

Comment:
Added a couple of more nullchecks here. Things can be null if there are syntax errors for example while typing `TestCaseSource(`, before typing the closing parenthesis.